### PR TITLE
Add conda -y check

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
Reference: https://github.com/rapidsai/ops/issues/2993

Add a check to ensure that all calls to conda use the -y/--yes flag
when appropriate to ensure non-interactivity.

### CODE REVIEW

File: pyproject.toml
Comment (bdice) [inline pyproject.toml:33]: Let's update this while we're at it. RAPIDS requires Python 3.9.
```suggestion
requires-python = ">=3.9"
```

File: pyproject.toml
Comment (bdice) [inline pyproject.toml:44]: Do you think this should be named something like `verify-conda-yes`? If I encountered a command `conda-yes` in the wild, I would expect it to invoke `conda --yes` as opposed to being a validator.

File: src/rapids_pre_commit_hooks/lint.py
Comment (bdice) [inline src/rapids_pre_commit_hooks/lint.py:40]: Let's avoid shadowing the builtin class `Warning`. Can we name this something like `ReplacementWarning`?

File: src/rapids_pre_commit_hooks/lint.py
Comment (bdice) [inline src/rapids_pre_commit_hooks/lint.py:78]: Let's raise an instance of the error class rather than the class itself.
```suggestion
                raise OverlappingReplacementsError(f"{r1} overlaps with {r2}")
```

File: src/rapids_pre_commit_hooks/lint.py
Comment (KyleFromNVIDIA) [reply src/rapids_pre_commit_hooks/lint.py]: Renamed to `LintWarning`.

**Comment by KyleFromNVIDIA** [comment]
`pre-commit run -a` passes.

File: pyproject.toml
Comment (KyleFromNVIDIA) [reply pyproject.toml]: Renamed to `verify-conda-yes`.

File: .github/workflows/run-tests.yaml
Comment (jameslamb) [inline .github/workflows/run-tests.yaml:18]:
```suggestion
        uses: actions/checkout@v4
```

The latest version of this action is `v4` (https://github.com/actions/checkout/releases).

I think we might as well go to the latest version of these third-party actions as we can, to extend as long as possible the time until they have to be upgraded again.

Comment (KyleFromNVIDIA) [reply .github/workflows/run-tests.yaml]: Updated to `actions/checkout@v4` and `actions/setup-python@v5`.

File: src/rapids_pre_commit_hooks/lint.py
Comment (bdice) [inline src/rapids_pre_commit_hooks/lint.py:85]: This could be consolidated into one line. The main benefit is that it's sorted on construction rather than being named as "sorted" while not yet being sorted.
```suggestion
        sorted_replacements = sorted([
            replacement
            for warning in self.warnings
            for replacement in warning.replacements
        ], key=lambda replacement: replacement.pos)
```

File: src/rapids_pre_commit_hooks/lint.py
Comment (bdice) [inline src/rapids_pre_commit_hooks/lint.py:94]: The reduction here is a little misleading because the reduction output isn't used.

We could alternatively write it with [pairwise](https://docs.python.org/3/library/itertools.html#itertools.pairwise), so that `check_overlap` isn't forced to return anything. (This will also do nothing if there are zero or one replacements, as desired.)
```python
from itertools import pairwise

for r1, r2 in pairwise(sorted_replacements):
    check_overlap(r1, r2)
```

Then you can remove `return r2` from `check_overlap`.

File: src/rapids_pre_commit_hooks/lint.py
Comment (bdice) [inline src/rapids_pre_commit_hooks/lint.py:97]: It might be nice to have a name for this:
```suggestion
        replaced_content = ""
```

**Review by bdice — APPROVED** [review]
If you'd like , we can merge this now to see if CI runs after merging `pr.yaml`, and iterate on any changes in a later PR.

**Comment by KyleFromNVIDIA** [comment]
@bdice Thanks, let's merge it now and see what happens.

**Pull request merged**
